ale/teacher/exploration/k_means.py

```diff
@@ -100,45 +78,47 @@ def __init__(self, corpus: Corpus, predictor: Predictor, seed: int, labels: List
-        self.clustered_documents = cluster_documents(corpus=corpus, k=self.k)
+        self.num_labels = len(self.labels)
+        self.clustered_documents = cluster_documents(corpus=corpus, num_labels=self.num_labels, seed=seed)
 
-    def propose(self, potential_ids: List[int], step_size: int,  budget: int) -> List[int]:
-        docs = self.clustered_documents.get_clustered_docs_by_idx(potential_ids)
-        clusters = self.clustered_documents.clusters
-        docs_per_cluster = int(step_size/len(clusters)) # equal distribution
-        output_ids: List[ClusterDocument] = []
-        empty_clusters = []
-
-        for cluster in clusters:
-            potential_docs_cluster = [doc for doc in docs if doc.cluster_idx==cluster]
-            potential_docs_cluster.sort(key=lambda x: x.distance, reverse=True)
-            if len(potential_docs_cluster) < docs_per_cluster: # less docs in cluster left than needed
-                output_ids.extend(potential_docs_cluster)
-                empty_clusters.append(cluster)
-            else:
-                output_ids.extend(potential_docs_cluster[:docs_per_cluster])
-
-        while step_size>len(output_ids) and len(empty_clusters)<len(clusters): # rest left
-            docs_per_rest_clusters = int((step_size-len(output_ids))/(len(clusters)-len(empty_clusters))) # equally distribute to not empty clusters
-            if docs_per_rest_clusters == 0: # less than 1 per empty cluster needed for stepsize reached
-                rest = step_size-len(output_ids)
-                for i in range(0,rest): # add 1 doc from first n not empty clusters (n: rest)
-                    not_empty_clusters = [cluster for cluster in clusters if cluster not in empty_clusters]
-                    curr_cluster = not_empty_clusters[i]
-                    potential_docs_cluster = [doc for doc in docs if doc.cluster_idx==curr_cluster]
-                    potential_docs_cluster.sort(key=lambda x: x.distance, reverse=True)
-                    output_ids.append(potential_docs_cluster[0])
-
-            for cluster in clusters:
-                if cluster not in empty_clusters:
-                    potential_docs_cluster = [doc for doc in docs if doc.cluster_idx==cluster]
-                    potential_docs_cluster.sort(key=lambda x: x.distance, reverse=True)
-                    if len(potential_docs_cluster) < docs_per_rest_clusters: # less docs in cluster left than needed
-                        output_ids.extend(potential_docs_cluster)
-                        empty_clusters.append(cluster)
-                    else:
-                        output_ids.extend(potential_docs_cluster[:docs_per_rest_clusters])
-        
-        out_ids = [item.idx for item in output_ids]
-        return out_ids
+    def propose(self, potential_ids: List[int], step_size: int, budget: int) -> List[int]:
+        pass#return propose_nearest_neighbors_to_centroids(self.clustered_documents, potential_ids, step_size, budget)
```

Comment:
Wieso pass?

Reply:
Unused strategy for the moment. Did not apply the refactoring to it. I will take a look at it now.

Reply:
Refactoring done. Please take a look at it.

Reply:
Looks good now